kcfinder/core/class/uploader.php

```diff
@@ -759,11 +768,19 @@ protected function callBack($url, $message="") {
                 $js = $this->$method($url, $message);
         }
 
-        if (!isset($js))
+        if ($this->outputFormat == 'json') {
+          header('Content-Type: application/json');
+          $json = $this->callBack_json($url, $message);
+          echo json_encode($json);
+        }
+        else {
+          if (!isset($js)) {
             $js = $this->callBack_default($url, $message);
 
-        header("Content-Type: text/html; charset={$this->charset}");
-        echo "<html><body>$js</body></html>";
+            header("Content-Type: text/html; charset={$this->charset}");
+            echo "<html><body>$js</body></html>";
```

Comment:
The above 2 lines should not be inside that conditional. The original version of this file only has 1 line "inside" the conditional (which is what happens when an if statement doesn't have brackets.